### PR TITLE
Check if lvmetad uses devices filter

### PR DESCRIPTION
**Pull request by ggkioulis**
In recent updates of lvm2 lvmetad does not seem to take into account the
devices/filter (Bug #1163526). A check has been added to ensure correct
functionality.

Two loop devices are attached to the same disk image, a device filter is created to counter this, and the Warnings from `pvs` are checked to see if if lvmetad reads the filter. If it does not, the test fails

- Related ticket: https://progress.opensuse.org/issues/63431
- Needles: No needles
- Verification run: 
sle-15-SP1: http://angmar.suse.de/tests/1507
tumbleweed: http://angmar.suse.de/tests/1494 

Note: Test has PASSED on tumbleweed and FAILED on sle-15-SP1, where the lvmetad functionality is not the expected.

### CODE REVIEW

**Comment by dzedro** [comment]
update also coverage ? :)

File: tests/console/lvm.pm [hunk 1]
```diff
@@ -123,4 +128,23 @@ sub run {
...
 }
 
+# Checks if lvmetad reads devices/filter (bsc #1163526)
+sub check_lvmetad_filter {
+    for (my $i = 1; $i <= 2; $i++) {
+        assert_script_run 'dd if=/dev/zero of=disk_' . $i . '.img bs=1M count=32 status=none';
+        assert_script_run 'losetup /dev/loop' . $i . ' disk_' . $i . '.img';
+        assert_script_run 'pvcreate /dev/loop' . $i;
+    }
+    assert_script_run 'losetup /dev/loop3 disk_2.img';
```
Comment (dzedro) [inline tests/console/lvm.pm:138]: Why is this here after the loop above ?

Comment (dzedro) [reply tests/console/lvm.pm:138]: TW test is passing not even noticing /dev/loop3, and 15sp1 is failing  on duplicate device.
That's some inconsistent behaviour or bug ?

Comment (ggkioulis) [reply tests/console/lvm.pm:138]: > Why is this here after the loop above ?

The 3rd losetup after the loop is there to attach a new loop device to disk_2.img, where a loop device is already attached to.

>TW test is passing not even noticing /dev/loop3, and 15sp1 is failing on duplicate device.
>That's some inconsistent behaviour or bug ?

https://bugzilla.suse.com/show_bug.cgi?id=1163526
It is a bug which appears in SLE but not tumbleweed, which is why I included the tumbleweed run. The inconsistency lies in the fact that --cache --config is passed my $filter, which removes the conflicting loop3 device, however the filter is not read in the newest lvm2 updates.

Comment (dzedro) [reply tests/console/lvm.pm:138]: Can you add workaround for bsc#1163526 ?

Comment (ggkioulis) [reply tests/console/lvm.pm:138]: Added soft fail

**Comment by dzedro** [comment]
passing on 15-SP2
https://openqa.suse.de/tests/4097037 x86_64
https://openqa.suse.de/tests/4097046 ppc64le
